### PR TITLE
fix(vite-node): bugfixes for production build

### PR DESCRIPTION
**Pull request by antfu**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt3/src/app/nuxt.ts [hunk 1]
```diff
@@ -116,6 +116,9 @@ export function createNuxtApp (options: CreateOptions) {
 
   // Expose runtime config
   if (process.server) {
+    options.ssrContext.runtimeConfig = options.ssrContext.runtimeConfig || {}
```
Comment (pi0) [inline packages/nuxt3/src/app/nuxt.ts:119]: Can we directly populate ssrContext with vite-node and avoid this workaround? This is unclear we are making a fix for vite-node only

Comment (antfu) [reply packages/nuxt3/src/app/nuxt.ts:119]: I just wasn't sure why we have this error. Do you have a reference where is supposed to be set so I can debug a bit.

File: packages/nuxt3/src/app/entry.async.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { CreateOptions } from '#app'
 
 const entry = process.server
   ? (ctx?: CreateOptions['ssrContext']) => import('#app/entry').then(m => m.default(ctx))
-  : () => import('#app/entry').then(m => m.default)
+  : () => import('#app/entry').then(m => m.default())
```
[the remark this answers is not in the document]
Comment (antfu) [reply packages/nuxt3/src/app/entry.async.ts:5]: Oh, I thought it was a typo (and somehow it did block vite-node), as the format does not match with L4.

File: packages/nuxt3/src/app/nuxt.ts (hunk 1 above)
Comment (pi0) [reply packages/nuxt3/src/app/nuxt.ts:119]: This should be normally filled at runtime by nitro: https://github.com/nuxt/framework/blob/9ebea904f4f3a7bd63b2a8ce5ecc5e592abf80c0/packages/nitro/src/runtime/app/render.ts#L90

File: packages/nuxt3/src/app/entry.async.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (pi0) [reply packages/nuxt3/src/app/entry.async.ts:5]: Yeah, little bit confusing with async entry. With client, we directly evaluate while with server, we let renderer do evaluate

File: packages/vite/src/runtime/server.mjs [hunk 3]
```diff
@@ -44,11 +44,11 @@ async function writeManifest (extraEntries) {
   await fs.writeFile(join(dir, 'client.manifest.mjs'), 'export default ' + JSON.stringify(clientManifest, null, 2), 'utf8')
 }
 
-export default (async () => {
+export default async (ssrContext) => {
   const { default: render } = await runner.executeFile(entry)
-  const result = await render()
+  const result = await render(ssrContext)
```
Comment (antfu) [inline packages/vite/src/runtime/server.mjs:49]: Now I understand - the root cause is this one, while I was debugging from the bottom up 😄

**Pull request merged**